### PR TITLE
update tutorial to use decoder contract

### PR DESCRIPTION
**Pull request by jmcook1186**
updates onchain stamps tutorial to use decoder contract

### CODE REVIEW

File: pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md [hunk 1]
```diff
@@ -208,206 +207,57 @@ The `chainID` for the network you are connected to is requested from the `provid
...
-  const uuid = await resolverContract.passports(address)
-}
-```
+Getting Passport data requires instantiating the `decoder` contract and calling its `getpassport` function. The `ethers` library provides everythign we need to instantiate the contract. Create a contract using `new etheres.Contract()` passing the contract address, ABI and the provider object as arguments. One complication is that the ABI is divided up into sections specific to each chain where the contract has been deployed, so you actually need to pass a specifier with the hex-encoded chain-ID too - for BaseGoerli this is `0x1a4`.
 
-It is also a good idea to add some error handling here, as it is possible that your connected user hasn't migrated any Stamps onchain yet, and may not have a `uuid` associated with their address. In this case, the returned `uuid` would be equal to the 32-byte hex encoded representation of 0:
+Once the contractinstance exists, you can simply call `getPassport` passing in the user address, which is stored in your app's state. If the function call returns some stamp data, you can set the `hasStampData` flag to `true` and return the data.
```
Comment (lebraat) [inline pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md:234]: space between contract/instance
Capitalize Stamp

**Review by lebraat — CHANGES_REQUESTED** [review]
Handful of comments! Thanks

File: pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md [hunk 2]
```diff
@@ -47,6 +47,8 @@ So your flow is:
 * pass the `uuid` to the EAS contract, returning an `Attestation`
 * decode and unpack the `Attestation,` returning the user's Stamp data
 
+Thankfully, there is now an onchaion shortcut to executing all this logicv. The `decoder` contract exposes an API that allows you to simply pass in an address and retrieve the decoded data, rather than having to execute the steps outlined above. Behind the scenes, the same logic is being followed, but you can abstract it away by itneracting with the `decoder` contract.
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md:50]: yes - done

File: pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md [hunk 3]
```diff
@@ -208,206 +207,57 @@ The `chainID` for the network you are connected to is requested from the `provid
...
-* The `resolver` contract is where you can request a `uuid` for an address.
-* The `EAS` contract is where you can pass a `uuid` and receive an `Attestation`.
-
-#### Additional files
+This is the addresses on the BaseGoerli blockchain where the `decoder` contract is stored. The `abi` is a formatted set of function signatures that allow the contract bytecode to be decoded and instantiated in your app.
 
-There are several elements being loaded into the app from local files. In this section, we will describe how to set these up. If you are building this app along with this tutorial as opposed to just using the example app, you will need to create these files and add the proper code to them, referenced in the links.
-
-These are the additional files that we will be creating:
+You will also need to load a file containing the weights for individual Stamps so you can calculate a Passport score.
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md:216]: I'm having issues with the `getScore` endpoint in the contract. While we're waiting for some debugging support from the dev team, the app still calculates the score client side, although the Stamps themselves are retrieved directly from the contract.

[the remark this answers is not in the document]
Comment (jmcook1186) [reply pages/building-with-passport/tutorials/integrating-onchain-stamp-data.md]: update - fixed the issue with getScore() - now the page is updated accordingly

**Pull request merged**